cpp/dolfin/mesh/cell_types.cpp

```diff
@@ -511,17 +513,43 @@ std::vector<std::uint8_t> mesh::vtk_mapping(mesh::CellType type, int num_nodes)
       throw std::runtime_error("Higher order tetrahedron not supported");
     }
   case mesh::CellType::quadrilateral:
-    switch (num_nodes)
-    {
-    case 4:
-    {
+  {
+    if (num_nodes == 4)
       // Assumes mapping from lexiographic ordering, not counter-clockwise
       // which is the VTK format
       return {0, 1, 3, 2};
+    {
+      int i = sqrt(num_nodes);
```

Comment:
This looks weird and isn't clear. Add a comment, and see if it can be done more clearly.

Reply:
Better now?

Reply:
Let's make it an assertion:
```
assert((sqrt(x) - floor(sqrt(x))) == 0); 
```

Reply:
Adding this assertion breaks the pytests, not sure why